### DISCUSSION TITLE
Lazy loading

### DISCUSSION THREAD

**Discussion by JanBN** [Comcast/react-data-grid#2433] [Q&A]
Is there some example with lazy loading. Or how to implement lazy loading ?

Thank you

**Comment by nstepien** [comment]
Can you clarify what is there to lazy load?

**Comment by JanBN** [comment]
I mean loading data from backend when you scroll down, or set some filters. E.g I have 100 000 rows by don't want to load them all since it's a lot of data and takes too long.

**Comment by nstepien** [comment]
Could try something like this with useSWR: https://swr.vercel.app/

```jsx
function LazyFormatter(props) {
  const { data } = useSWR(`/api/row/${props.row.id}`);

  if (!data) return null;

  return <>{data.name}</>;
}

function MyGrid() {
  const rows = [{ id: 1 }];

  return <DataGrid columns={columns} rows={rows} defaultColumnOptions={{ formatter: LazyFormatter }} />;
}
```

**Comment by amanmahajan7** [comment]
You can also use the `scroll` event and load as needed. Here is an example
https://github.com/adazzle/react-data-grid/blob/main/stories/demos/AllFeatures.tsx#L236
